File: flowforge-frontend/src/services/presenceService.js
import api from './api';

/**
 * Presence REST Client Service.
 * 
 * WHY THIS SERVICE EXISTS:
 * Intercepts HTTP requests to /api/v1/presence endpoints for querying online users list, counts, and status.
 */
export const presenceService = {
  // Get List of Online Users
  getOnlineUsers: async () => {
    const response = await api.get('/presence/online');
    return response.data.data;
  },

  // Get Online Count
  getOnlineCount: async () => {
    const response = await api.get('/presence/count');
    return response.data.data;
  },

  // Get User Presence Status
  getUserPresence: async (email) => {
    const response = await api.get(`/presence/users/${encodeURIComponent(email)}`);
    return response.data.data;
  },
};

export default presenceService;

File: flowforge-backend/src/main/java/com/flowforge/service/PresenceService.java
package com.flowforge.service;

import com.flowforge.dto.PresenceDto;

import java.util.List;

/**
 * Presence Service Interface.
 * 
 * WHY THIS INTERFACE EXISTS:
 * Defines business operations for tracking real-time WebSocket connection
 * sessions,
 * marking users online/offline, and querying online presence status.
 */
public interface PresenceService {

    void userConnected(String email);

    void userDisconnected(String email);

    List<PresenceDto> getOnlineUsers();

    PresenceDto getUserPresence(String email);

    long getOnlineCount();
}
